File: src/main/java/com/facimus/procesos/ejecucion/service/TableroService.java
package com.facimus.procesos.ejecucion.service;

import com.facimus.procesos.ejecucion.dto.response.TableroResponse;

/** Como va la operacion de una tienda: de un proceso o de todos. */
public interface TableroService {

    /**
     * El tablero de un proceso, o de la tienda entera si no se dice cual. Sale de la base en un numero fijo de
     * consultas agrupadas, crezcan los pedidos lo que crezcan.
     */
    TableroResponse de(Long empresaId, Long procesoId);
}

File: frontend/src/app/service/tablero.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { environment } from '../../environments/environment';
import { Tablero } from '../models/tablero.model';

/** Como va la operacion. Los mismos numeros para un proceso o para la tienda entera. */
@Injectable({ providedIn: 'root' })
export class TableroService {
  private readonly http: HttpClient = inject(HttpClient);
  private readonly api: string = `${environment.apiUrl}/api/v1`;

  /** El tablero de un proceso, o el de la tienda entera cuando no se dice cual. */
  obtener(procesoId: number | null): Observable<Tablero> {
    return procesoId === null
      ? this.http.get<Tablero>(`${this.api}/empresas/actual/tablero`)
      : this.http.get<Tablero>(`${this.api}/procesos/${procesoId}/tablero`);
  }
}
